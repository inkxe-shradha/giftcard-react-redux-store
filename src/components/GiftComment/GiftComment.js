import React from "react";
import StarRatingComponent from "react-star-rating-component";
import "../../pages/giftcard-details/index.scss";
import PropTypes from "prop-types";
import moment from "moment";
import { IconButton } from "@mui/material";
import DeleteIcon from "@mui/icons-material/Delete";
import ModeEditIcon from "@mui/icons-material/ModeEdit";
const GiftComment = ({ comment, currentUser, onRemove, onEdit }) => {
  return (
    <div className="comment-section">
      <p className="comment-name">
        {" "}
        <strong>
          {comment.name
            ? comment.name
            : `${comment.first_name} ${comment.last_name}`}
        </strong>
        <span className="comment-date">
          {" "}
          <i>{moment(comment.commented_on).format("MMM DD, YYYY")}</i>
        </span>
      </p>
      {/* Edit / Remove Button */}
      {currentUser && currentUser?.email === comment?.email && (
        <div className="float-right">
          <IconButton
            aria-label="Edit Comment"
            color="primary"
            onClick={() => onEdit(comment)}
          >
            <ModeEditIcon />
          </IconButton>
          <IconButton
            aria-label="Delete Comment"
            color="error"
            onClick={() => onRemove(comment.id)}
          >
            <DeleteIcon />
          </IconButton>
        </div>
      )}
      <p className="comment">{comment.comment}</p>
      <StarRatingComponent
        name="rating"
        emptyStarColor="#ffff"
        starCount={5}
        value={comment.rating}
        editing={false}
      />
    </div>
  );
};

// PropTypes
GiftComment.propTypes = {
  comment: PropTypes.object.isRequired,
  currentUser: PropTypes.object.isRequired,
  onRemove: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
};
export default GiftComment;
