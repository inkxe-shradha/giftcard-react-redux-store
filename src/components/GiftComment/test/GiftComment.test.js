import { render, screen } from "@testing-library/react";
import GiftComment from "../GiftComment";

describe("Gift card comment test suite", () => {
  const compProps = {
    comment: {
      id: 1,
      name: "John Doe",
      email: "jhondoe@gmail.com",
      comment: "This is a comment",
      rating: 4,
      commented_on: "2020-05-05T00:00:00.000Z",
    },
    currentUser: {},
    onRemove: jest.fn(),
    onEdit: jest.fn(),
  };

  it("Should render the comment component snapshot", () => {
    const { asFragment } = render(<GiftComment {...compProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should show the edit and delete button when commented used logged In", () => {
    render(
      <GiftComment
        {...compProps}
        currentUser={{ email: "jhondoe@gmail.com" }}
      />
    );
    expect(screen.getByLabelText("Edit Comment")).toBeInTheDocument();
    expect(screen.getByLabelText("Delete Comment")).toBeInTheDocument();
  });

  it("Should be contain the passed comment by the comment", () => {
    render(<GiftComment {...compProps} />);
    expect(screen.getByText("This is a comment")).toBeInTheDocument();
  });

  it("Should should be call the edit and delete function when edit and delete button clicked", () => {
    render(
      <GiftComment
        {...compProps}
        currentUser={{ email: "jhondoe@gmail.com" }}
      />
    );
    const editButton = screen.getByLabelText("Edit Comment");
    const deleteButton = screen.getByLabelText("Delete Comment");
    editButton.click();
    deleteButton.click();
    expect(compProps.onEdit).toHaveBeenCalled();
    expect(compProps.onRemove).toHaveBeenCalled();
  });
});
