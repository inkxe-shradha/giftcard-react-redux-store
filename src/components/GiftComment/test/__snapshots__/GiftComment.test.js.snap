// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift card comment test suite Should render the comment component snapshot 1`] = `
<DocumentFragment>
  <div
    class="comment-section"
  >
    <p
      class="comment-name"
    >
       
      <strong>
        John Doe
      </strong>
      <span
        class="comment-date"
      >
         
        <i>
          May 05, 2020
        </i>
      </span>
    </p>
    <p
      class="comment"
    >
      This is a comment
    </p>
    <div
      class="dv-star-rating dv-star-rating-non-editable"
      style="display: inline-block; position: relative;"
    >
      <input
        class="dv-star-rating-input"
        id="rating_5"
        name="rating"
        style="display: none; position: absolute; margin-left: -9999px;"
        type="radio"
        value="5"
      />
      <label
        class="dv-star-rating-star dv-star-rating-empty-star"
        for="rating_5"
        style="float: right; cursor: default; color: rgba(255, 255, 255, 1);"
      >
        <i
          style="font-style: normal;"
        >
          ★
        </i>
      </label>
      <input
        checked=""
        class="dv-star-rating-input"
        id="rating_4"
        name="rating"
        style="display: none; position: absolute; margin-left: -9999px;"
        type="radio"
        value="4"
      />
      <label
        class="dv-star-rating-star dv-star-rating-full-star"
        for="rating_4"
        style="float: right; cursor: default; color: rgb(255, 180, 0);"
      >
        <i
          style="font-style: normal;"
        >
          ★
        </i>
      </label>
      <input
        class="dv-star-rating-input"
        id="rating_3"
        name="rating"
        style="display: none; position: absolute; margin-left: -9999px;"
        type="radio"
        value="3"
      />
      <label
        class="dv-star-rating-star dv-star-rating-full-star"
        for="rating_3"
        style="float: right; cursor: default; color: rgb(255, 180, 0);"
      >
        <i
          style="font-style: normal;"
        >
          ★
        </i>
      </label>
      <input
        class="dv-star-rating-input"
        id="rating_2"
        name="rating"
        style="display: none; position: absolute; margin-left: -9999px;"
        type="radio"
        value="2"
      />
      <label
        class="dv-star-rating-star dv-star-rating-full-star"
        for="rating_2"
        style="float: right; cursor: default; color: rgb(255, 180, 0);"
      >
        <i
          style="font-style: normal;"
        >
          ★
        </i>
      </label>
      <input
        class="dv-star-rating-input"
        id="rating_1"
        name="rating"
        style="display: none; position: absolute; margin-left: -9999px;"
        type="radio"
        value="1"
      />
      <label
        class="dv-star-rating-star dv-star-rating-full-star"
        for="rating_1"
        style="float: right; cursor: default; color: rgb(255, 180, 0);"
      >
        <i
          style="font-style: normal;"
        >
          ★
        </i>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
